otelconf/log.go

```diff
@@ -0,0 +1,236 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelconf // import "go.opentelemetry.io/contrib/otelconf"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutlog"
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/noop"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"google.golang.org/grpc/credentials"
+
+	"go.opentelemetry.io/contrib/otelconf/internal/tls"
+)
+
+func loggerProvider(cfg configOptions, res *resource.Resource) (log.LoggerProvider, shutdownFunc, error) {
+	provider, ok := cfg.opentelemetryConfig.LoggerProvider.(*LoggerProviderJson)
+	if provider == nil {
+		return noop.NewLoggerProvider(), noopShutdown, nil
+	}
+	if !ok {
+		return noop.NewLoggerProvider(), noopShutdown, newErrInvalid("logger_provider")
+	}
```

Comment:
`provider` will always be `nil` when `ok` is `false`.

```suggestion
	if !ok {
		return noop.NewLoggerProvider(), noopShutdown, newErrInvalid("logger_provider")
	}
	if provider == nil {
		return noop.NewLoggerProvider(), noopShutdown, nil
	}
```